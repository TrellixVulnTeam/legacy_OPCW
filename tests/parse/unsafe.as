# Some random unsafe block.
unsafe {  }

# Declare a mutable static.
# static mut data_race: int = 0;

# Access the data_race.
unsafe {
    data_race += 21;
}
