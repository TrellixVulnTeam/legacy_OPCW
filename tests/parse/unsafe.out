ModuleDecl <?> _
  Nodes <?> 
    UnsafeBlock <?>
      Nodes <?> 
    StaticSlotDecl <?> mut data_race 'int'
      IntegerExpr <?> 'int' 0 (10)
    UnsafeBlock <?>
      Nodes <?> 
        AssignAddExpr <?>
          Ident <?> data_race
          IntegerExpr <?> 'int' 21 (10)
